### ISSUE TITLE
Join without using any primary key

### ISSUE THREAD

**Issue by awm33** [balderdashy/waterline-schema#19]
I have two models from a legacy database with integer primary keys that I'd like to associate using string columns that match. I can't get them to associate. It seems to assume the primary key on at least one side

 	 {
		tableName: 'domains',
		identity: 'domain',
		attributes: {
			id: {
				type: 'integer',
				primaryKey: true,
				autoIncrement: true
			},
			domain: {
				type: 'string',
				required: true
			},
			account: {
				model: 'account',
				columnName: 'domain',
				type: 'string'
			}
			...
		}
	}

 	 {
		tableName: 'accounts',
		identity: 'account',
		attributes: {
			id: {
				type: 'integer',
				primaryKey: true,
				autoIncrement: true
			},
			domain: {
				model: 'domain',
				columnName: 'domain',
				type: 'string'
			}
			...
		}
	}

**Comment by devinivy** [comment]
Try using the via attribute to specify the foreign key and let us know how it goes!

```
{
    tableName: 'domains',
    identity: 'domain',
    attributes: {
        id: {
            type: 'integer',
            primaryKey: true,
            autoIncrement: true
        },
        domain: {
            type: 'string',
            required: true
        },
        account: {
            model: 'account',
            via: 'domain'
        }
        ...
    }
}

 {
    tableName: 'accounts',
    identity: 'account',
    attributes: {
        id: {
            type: 'integer',
            primaryKey: true,
            autoIncrement: true
        },
        domain: {
            model: 'domain',
            via: 'account'
        }
        ...
    }
}
```

**Comment by awm33** [comment]
I had tried `via` before (I should of put that) but it looks like all combinations of `via` with account/domain, domain/domain, account/account, domain/account

As suggested I get a `ER_BAD_FIELD_ERROR: Unknown column 'domain.account' in 'field list'` which means it's passing it to mysql

**Comment by tjwebb** [comment]
Have you tried using `on`? `on` is an internal key used to link up the fields, but isn't a documented part of the public API currently.

**Comment by awm33** [comment]
I just tried `on` with both the column names and model attribute names. I also tried adding `references` trying both the model identity and table names

**Comment by devinivy** [comment]
I think the confounding factor is that you're using both a column name and an association.  If the field name matches the column name and the foreign `via` attribute, does this work for you?

**Comment by awm33** [comment]
So if I set both the attribute name and `via` to the database column name. It doesn't error out, but it doesn't populate

    domain: {
        model: 'domain',
        via: 'domain'
    }

    domain: {
        model: 'account',
        via: 'domain'
    }

**Comment by devinivy** [comment]
It does look like waterline-schema requires that the join be on the primary key.
https://github.com/balderdashy/waterline-schema/blob/master/lib/waterline-schema/joinTables.js#L314-L331

**Comment by sailsbot** [comment]
Thanks for posting, @awm33.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
